src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs

```diff
@@ -255,5 +256,50 @@ public async Task<ActionResult> DeleteOrderAsync(string orderId)
             await _orderRepository.UpdateOrderAsync(order);
             return NoContent();
         }
+
+        [HttpPut]
+        [Route("{orderId}/status")]
+        public async Task<ActionResult<ErrorResponseModel>> UpdateStatusAsync(string orderId, StatusModel model)
+        {
+            if (model is null)
+            {
+                throw new ArgumentNullException(nameof(model));
+            }
+
+            var order = await _orderRepository.GetOrderByIdAsync(orderId);
+
+            if (order is null)
+            {
+                return NotFound();
+            }
+
+            var primaryOrganisationId = User.GetPrimaryOrganisationId();
+            if (primaryOrganisationId != order.OrganisationId)
+            {
+                return Forbid();
+            }
+
+            int serviceRecipientsCount = await _serviceRecipientRepository.GetCountByOrderIdAsync(orderId);
+            if (!order.IsSectionStatusComplete(serviceRecipientsCount))
+            {
+                var createOrderResponse = new ErrorResponseModel();
+                createOrderResponse.Errors = new[] { new ErrorModel("OrderNotComplete", "Order") };
+                return BadRequest(createOrderResponse);
+            }
+
+            var enumParsed = Enum.TryParse<OrderStatus>(model.Status, out var orderStatus);
+            if (!enumParsed)
+            {
+                throw new InvalidEnumArgumentException(nameof(model.Status));
+            }
```

Comment:
Is it possible to use an `enum` for the `StatusModel.Status` to avoid having to do this?

Reply:
It messes up the validation unfortunately, we'd need to look into some sort of actionfilter and I'd rather not do that just now